cmd/wait-for-github/pr_test.go

```diff
@@ -0,0 +1,262 @@
+	return fg.MergedCommit, fg.Closed, fg.MergedAt, fg.Error
+}
+
+func TestPRCheck(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name             string
+		fakeClient       FakeGithubClientPRCheck
+		err              error
+		expectedExitCode *int
+	}{
+		{
+			name: "PR is merged",
+			fakeClient: FakeGithubClientPRCheck{
+				MergedCommit: "abc123",
+				MergedAt:     1234567890,
+			},
+			expectedExitCode: &zero,
+		},
+		{
+			name: "PR is closed",
+			fakeClient: FakeGithubClientPRCheck{
+				Closed: true,
+			},
+			expectedExitCode: &one,
+		},
+		{
+			name: "PR is open",
+			fakeClient: FakeGithubClientPRCheck{
+				MergedCommit: "",
+				Closed:       false,
+			},
+			expectedExitCode: &one,
+		},
+		{
+			name: "Error from IsPRMergedOrClosed",
+			fakeClient: FakeGithubClientPRCheck{
+				Error: fmt.Errorf("an error occurred"),
+			},
+			err: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakePRStatusChecker := &tt.fakeClient
+			cfg := &config{
+				recheckInterval: 1,
+			}
+
+			ctx, cancel := context.WithTimeout(context.Background(), 1)
+			cancel()
+
+			prConfig := prConfig{
+				owner: "owner",
+				repo:  "repo",
+				pr:    1,
+			}
+
+			err := checkPRMerged(ctx, fakePRStatusChecker, cfg, &prConfig)
+			if tt.expectedExitCode != nil {
+				var exitErr cli.ExitCoder
+				require.ErrorAs(t, err, &exitErr)
+				require.Equal(t, *tt.expectedExitCode, exitErr.ExitCode())
+			} else if err != nil {
+				require.ErrorAs(t, err, &tt.err)
+			}
+		})
+	}
+}
+
+type fakeFileWriter struct {
+	filename string
+	data     []byte
+	perm     fs.FileMode
+
+	err error
+}
+
+func (f *fakeFileWriter) WriteFile(filename string, data []byte, perm fs.FileMode) error {
+	f.filename = filename
+	f.data = data
+	f.perm = perm
+
+	return f.err
+}
+
+func TestWriteCommitInfoFile(t *testing.T) {
+	prConfig := prConfig{
+		owner: "owner",
+		repo:  "repo",
+		pr:    1,
+
+		commitInfoFile: "commit_info_file",
+		writer:         &fakeFileWriter{},
+	}
+
+	prCheck := &prCheck{
+		prConfig: prConfig,
+
+		githubClient: &FakeGithubClientPRCheck{
+			MergedCommit: "abc123",
+			MergedAt:     1234567890,
+		},
+	}
+
+	err := prCheck.Check(context.TODO(), 1)
+	var cliExitErr cli.ExitCoder
+	require.ErrorAs(t, err, &cliExitErr)
+	require.Equal(t, 0, cliExitErr.ExitCode())
+
+	ffw := prConfig.writer.(*fakeFileWriter)
+
+	var gotCommitInfo commitInfo
+	err = json.Unmarshal(ffw.data, &gotCommitInfo)
+	require.NoError(t, err)
+
+	require.Equal(t, "commit_info_file", ffw.filename)
+	require.Equal(t, commitInfo{
+		Owner:    "owner",
+		Repo:     "repo",
+		Commit:   "abc123",
+		MergedAt: 1234567890,
+	}, gotCommitInfo)
+	require.Equal(t, fs.FileMode(0644), ffw.perm)
+}
+
+type erroringFileWriter struct{}
+
+func (e erroringFileWriter) WriteFile(filename string, data []byte, perm fs.FileMode) error {
+	return fmt.Errorf("error")
+}
```

Comment:
Super nit, but I think you could accomplish the same by just defining `erroringFileWriter` as a `fakeFileWriter` with an error. Reduces # lines of code slightly. Something like:

```suggestion
var erroringFileWriter := &fakeFileWriter{err:  fmt.Errorf("error")}
```

Reply:
Super nice, that works, I will change it, thanks!

Reply:
https://github.com/grafana/wait-for-github/compare/51e6764f27800fce2c9c7439f2fab6b7ad257147..356feac05a700ff6275254f2c1a9fe57f50867d6

going to merge now :muscle: